### ISSUE TITLE
Replace inconsistent InspectorControlsTabsHint with a Notice

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#54689]
### Description

In teh Site editor and in the Post editor, two very similar components are used to inform users that some functionalities have moced to anoterh spot in the Editor. These two components serve the same purpose but:
- They are visually inconsistent.
- The use two separete components: Notice vs. InspectorControlsTabsHint

Screenshot:

<img width="713" alt="Screenshot 2023-09-21 at 11 40 23" src="https://github.com/WordPress/gutenberg/assets/1682452/dc96292c-a6db-4c1c-8441-22d50eabaab4">

te textual content (including the non visible one) of these notice and hint is, respectively:

```
Information notice
Looking for template parts? Find them in "Patterns".
Close
```

and 

```
Looking for other block settings? They've moved to the styles tab.
Dismiss hint
```

I'm not sure why from a design perspective the Post editor does not use a Notice. To me, this inconsistency seems less than ideal and I'm not sure about the reasoning behind using a 'hint` while the most appropriate UI pattern would be a Notice.

Moreover:
- The `InspectorControlsTabsHint` is an ad-hoc implementation and only used for this specific case in the Post editor.
- The `InspectorControlsTabsHint` has the same accessibility issues that have been recently addressed for the Notice omponent in https://github.com/WordPress/gutenberg/pull/54498

Noting that the Close button of the `InspectorControlsTabsHint` component doesn't use a Tooltip. This is one more vase where an icon button doesn't visually expose its accessible name. We use Tooltips for that specific reason: icon buttons _must_ visually expose their accessible name and worth reminding a tooltip is already a compromise from an accessibility perspective.

Since this is yet another case of icon button without a Tooltip, I'd really like to see this enforced via code in the Button component. Cc @ciampo 

Overall I'm not sure I see any good reason to use the `InspectorControlsTabsHint` component. It should be replaced with a Notice of typo `info`.

### Step-by-step reproduction instructions

- Use a clean database or make sure the preference  to delete the wp_usermeta > wp_persisted_preferences user meta
- If using the Gutenberg wp-env, you mahy want to try `wp user meta delete <your_user> wp_persisted_preferences`
- Go to the Site editor.
- Observe the info notice int eh navigation panel.
- Go tto the Post editor.
- Observe the 'hint' in the Settings panel
- Observe they are visually inconsistent.
- Observe the Close button if the hint in the post editor doesn't show a tooltip

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by ciampo** [comment]
> Since this is yet another case of icon button without a Tooltip, I'd really like to see this enforced via code in the Button component. Cc @ciampo

Thank you for the ping — this isn't the first time I get pinged around icons and tooltips.

As explained before, while I agree that we should follow accessibility guidelines, enforcing the change at the code level (ie. in this case on the `Button` component) would introduce many unexpected changes in the editor, and is simply not the way to go when collaborating on a project.

The real conversation to be had is at the design / UX level; **we should understand if we're good with always showing tooltips for icon-only buttons, and otherwise find accessible and suitable alternatives**. Only once we agreed on a way forward, code changes should be made according to the decisions made.

cc @WordPress/gutenberg-design

**Comment by jasmussen** [comment]
I agree, they need to be synced up. The bigger challenge here is that where we should move is in the direction of the rightmost one. Color to indicate warning level is not going to be that useful to folks with color blindness, and the usual stoplight colors are not always culturally indicative of good or bad across the world. In that light, I'd like to see a refresh of the Notice component to be more like the indicator on the right by default.
